style/ruby/README.md

```diff
@@ -659,4 +662,498 @@ end
+```
+
+## Formato de Palavra-Chave para Comentários
+
+A palavra-chave de anotação é seguida por dois pontos e um espaço, depois uma nota descrevendo o problema.
+
+```ruby
+# ruim
+def bar
+  # FIXME This has crashed occasionally since v3.2.1.
+  baz(:quux)
+end
+
+# bom
+def bar
+  # FIXME: This has crashed occasionally since v3.2.1.
+  baz(:quux)
+end
+```
+
+Se forem necessárias várias linhas para descrever o problema, as linhas subsequentes devem ser recuadas três espaços após o # (um geral mais dois para fins de recuo).
+
+```ruby
+def bar
+  # FIXME: This has crashed occasionally since v3.2.1. It may
+  #   be related to the BarBazUtil upgrade.
+  baz(:quux)
+end
+```
+
+as palavras-chaves são: `TODO`, `FIXME`, `OPTIMIZE`, `HACK`, `REVIEW`
+
+## TODO
+Use `TODO` para observar recursos ou funcionalidades ausentes que devem ser adicionados em uma data posterior.
+
+## FIXME
+Use `FIXME` para observar código quebrado que precisa ser corrigido.
+
+## OPTIMIZE
+Use `OPTIMIZE` para observar código lento ou ineficiente que pode causar problemas de desempenho.
+
+## HACK
+Use `HACK` para observar "code smells" onde práticas de codificação questionáveis foram utilizadas e devem ser refatoradas.
+
+## REVIEW
+Use `REVIEW` para observar qualquer coisa que deve ser verificada para confirmar se está funcionando conforme o pretendido. Por exemplo: 
+```ruby
+# REVIEW: Are we sure this is how the client does X currently?
+```
+
+***Use outras palavras-chave personalizadas de anotação se parecer apropriado, mas certifique-se de documentá-las no README ou em um documento semelhante do projeto.***
+
+# Blocks
+
+## Delimitadores de Blocos de Linha Única
+```ruby
+names = %w[Bozhidar Filipp Sarah]
+
+# ruim
+names.each do |name|
+  puts name
+end
+
+# bom
+names.each { |name| puts name }
+
+# ruim
+names.select do |name|
+  name.start_with?('S')
+end.map { |name| name.upcase }
+
+# bom
+names.select { |name| name.start_with?('S') }.map(&:upcase)
+```
+
+## Abreviação de Aplicação de Proc '`&:`'
+
+Use a abreviação de chamada `Proc` quando o método chamado é a única operação de um bloco.
+
+```ruby
+# ruim
+bluths.map { |bluth| bluth.occupation }
+bluths.select { |bluth| bluth.blue_self? }
+
+# bom
+bluths.map(&:occupation)
+bluths.select(&:blue_self?)
+```
+
+# Metodos
+
+## Método Perigoso com Sinal de Exclamação (bang)
+
+Os nomes de métodos potencialmente perigosos (ou seja, métodos que modificam o objeto ou seus argumentos, `exit!` (que não executa os finalizadores como o `exit` faz), etc) devem terminar com um ponto de exclamação se existir uma versão segura desse método perigoso.
+
+```ruby
+# ruim - Não há nenhum método 'seguro' correspondente.
+class Person
+  def update!
+  end
+end
+
+# bom
+class Person
+  def update
+  end
+end
+
+# bom
+class Person
+  def update!
+  end
+
+  def update
+  end
+end
+```
+
+## Defina o seguro usando o inseguro
+
+Defina o método não-bang (seguro) em termos do método bang (perigoso) sempre que possível.
```

Comment:
>Essa é uma implementação de um serviço `MyService` que possui dois métodos `call!` e `call`.
O método `call!` é considerado perigoso, pois pode lançar uma exceção `ActiveRecord::RecordNotFound` se não encontrar um registro com o `id` fornecido no banco de dados através do método `find`.
O método `call` é considerado seguro, pois não lança a exceção acima. Ele simplesmente chama o método `call!` com o mesmo argumento `id` e, em seguida, captura a exceção `ActiveRecord::RecordNotFound` usando uma cláusula `rescue`. Nesse caso, a exceção capturada é tratada retornando `nil`.
O objetivo de definir o método `call` em termos do método `call!` é que isso reduz a quantidade de código duplicado necessário para tratar a exceção `ActiveRecord::RecordNotFound`. Ao invés de escrever o tratamento da exceção duas vezes, uma no método `call!` e outra no método `call`, a implementação coloca o tratamento no método `call!` e simplesmente reaproveita essa implementação no método `call`.
Dessa forma, os usuários do serviço `MyService` podem chamar o método `call` sem se preocupar em lidar com a exceção `ActiveRecord::RecordNotFound`, pois isso já foi tratado internamente pela implementação do serviço.
https://rubystyle.guide/#safe-because-unsafe